src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -329,9 +329,18 @@
 
   :handle-ok
   (fn [context]
-    (->
-      (ca/get-certificate-statuses settings)
-      (as-json-or-pson context))))
+    (let [query-params (:params request)
+          queried-state (get query-params "state")]
+      (->
+        (case queried-state
+          "requested" (ca/filter-requested-certificates settings)
+          "signed" (ca/filter-signed-certificates settings)
+          "revoked" (ca/filter-revoked-certificates settings)
+          ;; This will return all statuses if someone requests a state that we
+          ;; don't know about. Is that desirable? Or should we only return all
+          ;; statuses if no state is requested?
+          (ca/get-certificate-statuses settings))
```

Comment:
Seems like opinion is split on whether to silently ignore bad query params or raise, for example, a 400 Bad Request with a useful message (perhaps including a list of valid states). I feel like the second is probably better UX, but on the other hand, we probably don't really expect people to be using this by hand much.

Reply:
yeah, I agree that a message is better UX, but figured I'd ask before investigating how to raise 😅

Reply:
the other reason I didn't just add it is because, currently, if you supply _any_ query params, they are happily ignored.

Reply:
Hmm... wonder if this becomes a backwards-compatability question then.

Reply:
yeah, should we ask Nick about it or..?

Reply:
No, he probably won't know. @pcarlisle or @justinstoller might have feelings though.

Reply:
I don't really know if anyone is sending anything else right now. I guess I'd lean to not worrying about it and continue to allow whatever but possibly put something in for the next major version.